xtask/src/dist.rs

```diff
@@ -58,30 +58,67 @@ fn dist_client(version: &str, release_tag: &str) -> Result<()> {
 }
 
 fn dist_server() -> Result<()> {
-    if cfg!(target_os = "linux") {
+    let target = get_target();
+    if target.contains("-linux-gnu") {
         env::set_var("CC", "clang");
     }
-    cmd!("cargo build --manifest-path ./crates/rust-analyzer/Cargo.toml --bin rust-analyzer --release").run()?;
-
-    let (src, dst) = if cfg!(target_os = "linux") {
-        ("./target/release/rust-analyzer", "./dist/rust-analyzer-linux")
-    } else if cfg!(target_os = "windows") {
-        ("./target/release/rust-analyzer.exe", "./dist/rust-analyzer-windows.exe")
-    } else if cfg!(target_os = "macos") {
-        ("./target/release/rust-analyzer", "./dist/rust-analyzer-mac")
-    } else {
-        panic!("Unsupported OS")
-    };
-
-    let src = Path::new(src);
-    let dst = Path::new(dst);
 
+    let toolchain = get_toolchain(&target);
+    cmd!("cargo +{toolchain} build --manifest-path ./crates/rust-analyzer/Cargo.toml --bin rust-analyzer --target {target} --release").run()?;
+    let src =
+        Path::new("target").join(format!("rust-analyzer{}", get_executable_extension(&target)));
+    let dst = Path::new("dist").join(get_release_artifact_name(&target));
     cp(&src, &dst)?;
     gzip(&src, &dst.with_extension("gz"))?;
 
     Ok(())
 }
 
+fn get_target() -> String {
+    match env::var("RA_TARGET") {
+        Ok(target) => target,
+        _ => {
+            if cfg!(target_os = "linux") {
+                "x86_64-unknown-linux-gnu".to_string()
+            } else if cfg!(target_os = "windows") {
+                "x86_64-pc-windows-msvc".to_string()
+            } else if cfg!(target_os = "macos") {
+                "x86_64-apple-darwin".to_string()
+            } else {
+                panic!("Unsupported OS")
+            }
+        }
+    }
+}
+
+fn get_executable_extension(target: &str) -> String {
```

Comment:
```suggestion
fn get_executable_suffix(target: &str) -> String {
```

https://doc.rust-lang.org/std/env/consts/constant.EXE_SUFFIX.html

extensioni s a thing without `.`.